ambry-router/src/main/java/com/github/ambry/router/SimpleOperationTracker.java

```diff
@@ -337,25 +331,24 @@ class SimpleOperationTracker implements OperationTracker {
 
     int numActiveReplicasInOriginatingDc =
         getEligibleReplicas(originatingDcName, EnumSet.of(ReplicaState.STANDBY, ReplicaState.LEADER)).size();
-    if (routerConfig.routerOperationTrackerTerminateOnNotFoundEnabled && numReplicasInOriginatingDc > 0
+    if (routerConfig.routerOperationTrackerTerminateOnNotFoundEnabled
         && numActiveReplicasInOriginatingDc >= routerConfig.routerPutSuccessTarget) {
-      // This condition accounts for following cases:
-      // 1. Intermediate state of moving replicas (there could be 6 replicas in originating dc temporarily).
-      // 2. Looks at all replicas (instead of routerPutSuccessTarget) in originating DC since one of the replicas
-      // in which the blob was PUT originally could be in error state or is being rebuilt from scratch. By looking at
-      // all replicas, we make sure that we don't miss the other replica in which blob was PUT.
-      // 3. Uses this feature only if there are at least 'routerPutSuccessTarget' number of replicas in leader or
-      // standby state. This ensures that all the replicas in PUT quorum are not down (or being rebuilt) which means
-      // that there is at least one active replica containing the blob.
-      originatingDcNotFoundFailureThreshold = numReplicasInOriginatingDc;
+      // There are two conditions to meet in order to use this feature
```

Comment:
Just to confirm,
If routerOperationTrackerTerminateOnNotFoundEnabled, with the logic, it'll fix the 503 error.
If it's not enabled, in the case(8 replica returned not found and 1 returned not available), we'll still see 503, right?

Reply:
that's right.